src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -5661,19 +5661,12 @@ VOID DECLSPEC_NORETURN DispatchManagedException(OBJECTREF throwable, CONTEXT* pE
 
     Thread *pThread = GetThread();
 
-    if (preserveStackTrace)
-    {
-        pThread->IncPreventAbort();
-        ExceptionPreserveStackTrace(throwable);
```

Comment:
I am wondering whether the one remaining use of `ExceptionPreserveStackTrace` in the VM is actually necessary. It is wrapped in very confusing logic that does not look right and we seem to be getting lucky that it never kicks in practice.

It is fine to do clean it up in a separate PR.

Reply:
Thank you, I'll look into it separately.